src/main/kotlin/no/skatteetaten/aurora/boober/service/DeployService.kt

```diff
@@ -79,11 +78,25 @@ class DeployService(
         val deployId = UUID.randomUUID().toString()
 
         val openShiftResponses: List<OpenShiftResponse> = performOpenShiftCommands(deployId, deploymentSpec)
-        val performDeploy = !(deploymentSpec.deploy?.flags?.pause ?: false) && shouldDeploy
-        val redeployResponse: ResponseEntity<JsonNode>? = if (performDeploy) triggerRedeploy(deploymentSpec, openShiftResponses) else null
 
         val success = openShiftResponses.all { it.success }
-        return AuroraDeployResult(deployId, deploymentSpec, openShiftResponses, redeployResponse, success)
+        val noPause = !(deploymentSpec.deploy?.flags?.pause ?: false)
+        val performDeploy = success && noPause && shouldDeploy
+
+        val tagResult = if (performDeploy) {
+            deploymentSpec.deploy?.takeIf { it.releaseTo != null }?.let {
+                val dockerGroup = it.groupId.replace(".", "_")
+                val cmd = TagCommand("${dockerGroup}/${it.artifactId}", it.version, it.releaseTo!!, dockerRegistry)
+                val response = dockerService.tag(cmd)
+                TagResult(cmd, response, response.statusCode.is2xxSuccessful)
+            }
+        } else {
+            null
+        }
+        val redeployResponse = if (performDeploy) triggerRedeploy(deploymentSpec, openShiftResponses) else null
+
+        val totalSuccess = success && tagResult?.success ?: true && redeployResponse?.success ?: true
```

Comment:
This is a mouthful; the name of the success variable has become to general in the new context and should be renamed and the rest of the expression is too complex to read easily. Maybe something like

        val isTrue = { b : Boolean -> b }
        val totalSuccess = listOf(
                success,
                tagResult?.success ?: true,
                redeployResponse?.success ?: true
        ).all(isTrue)